libethereum/ClientTest.cpp

```diff
@@ -61,6 +61,7 @@ void ClientTest::setChainParams(string const& _genesis)
         if (params.sealEngineName != "NoProof")
             BOOST_THROW_EXCEPTION(ChainParamsNotNoProof() << errinfo_comment("Provided configuration is not well formatted."));
 
+        m_signalled.notify_all(); // to wake up the thread from Client::doWork()
```

Comment:
Can we move this to `Client::reopenChain(ChainParams const& _p, WithExisting _we)` before `stopWorking()` call?